src/tiramisu_function.cpp

```diff
@@ -1634,7 +1731,19 @@ void tiramisu::function::lift_mpi_comp(tiramisu::computation *comp) {
     }
 }
 
-void tiramisu::function::codegen(const std::vector<tiramisu::buffer *> &arguments, const std::string obj_filename, const bool gen_cuda_stmt) {
+void tiramisu::function::codegen(const std::vector<tiramisu::buffer *> &arguments, const std::string obj_filename, const bool gen_cuda_stmt) 
+{
+    if (gen_cuda_stmt) 
+    {
+	if(!this->mapping.empty())
```

Comment:
You should check also if get_automatic_gpu_copy() return true.

Reply:
no, it is specific to a buffer, it is not a global fied
i did that to be able to mix manual and automatic , each buffer is independent